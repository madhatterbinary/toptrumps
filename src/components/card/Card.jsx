import React from 'react';
import PropTypes from 'prop-types';
import CardHeader from './CardHeader';
import CardImage from './CardImage';
import CardAttributeList from './CardAttributeList';

const Card = (props) => {
  const {
    isClosed,
    card,
    handleAttributeSelection,
    status,
    selectedAttribute,
    isWinner
  } = props;

  return (
    <div className={`card ${isClosed && 'closed'}`}>
      {!isClosed ? (
        <div className="card-body">
          <CardHeader
            title={card.name}
            model={card.model}
            manufacturer={card.manufacturer}
          />
          <CardImage
            image={card.image}
          />
          <CardAttributeList
            card={card}
            handleAttributeSelection={handleAttributeSelection}
            status={status}
            selectedAttribute={selectedAttribute || ""}
            isWinner={isWinner}
          />
        </div>
      ) : null}
    </div>
  );
}

Card.propTypes = {
  isClosed: PropTypes.bool.isRequired,
  card: PropTypes.object.isRequired,
  handleAttributeSelection: PropTypes.func.isRequired,
  status: PropTypes.string.isRequired,
}

export default Card;
