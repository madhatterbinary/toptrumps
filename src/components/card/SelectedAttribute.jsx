import React from 'react';
import PropTypes from 'prop-types';
import attributes from '../../constants/attributes';

const SelectedAttribute = (props) => {
  const { 
    selectedAttribute,
    attributeValue,
    isClosed,
    isWinner
  } = props;
  return selectedAttribute && isClosed ? (
    <div className="selected-attribute">
    {isWinner ?
      <span className="badge badge-success">{attributes[selectedAttribute]}:
    {attributeValue}</span> :
     <span>    {attributes[selectedAttribute]}:
    {attributeValue}</span>
    }
    </div>
  ) : (
    <div className="selected-attribute">Select an attribute</div>
  );
}

SelectedAttribute.propTypes = {
  selectedAttribute: PropTypes.string,
  attributeValue: PropTypes.string,
}

export default SelectedAttribute;
