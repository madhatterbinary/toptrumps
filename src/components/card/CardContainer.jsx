import React from 'react';
import PropTypes from 'prop-types';
import Card from './Card';
import SelectedAttribute from './SelectedAttribute';

const CardContainer = (props) => {
  const {
    card,
    isClosed,
    isWinner,
    selectedAttribute,
    handleAttributeSelection,
    status,
  } = props;

  return (
    <div className={`card-container ${isWinner && 'winner'}`}>
      <Card 
        card={card}
        isClosed={isClosed}
        handleAttributeSelection={handleAttributeSelection}
        status={status}
        selectedAttribute={selectedAttribute}
      />
      {!isClosed ? (
      <SelectedAttribute
        selectedAttribute={selectedAttribute}
        attributeValue={card[selectedAttribute]}
        isClosed
        isWinner={isWinner}
      />
      ) : null}
      <div className="card-result">You won!</div>
    </div>
  );
}

CardContainer.propTypes = {
  card: PropTypes.object.isRequired,
  isClosed: PropTypes.bool.isRequired,
  isWinner: PropTypes.bool.isRequired,
  status: PropTypes.string.isRequired,
  selectedAttribute: PropTypes.string,
  handleAttributeSelection: PropTypes.func.isRequired,
}

export default CardContainer;
