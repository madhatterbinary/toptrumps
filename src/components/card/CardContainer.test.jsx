import React from 'react';
import CardContainer from './CardContainer';
import SelectedAttribute from './SelectedAttribute';
import Card from './Card';
import TestRenderer from 'react-test-renderer';

describe('CardContainer', () => {
  it('renders', () => {
    const component = TestRenderer.create(<CardContainer card={{}} handleAttributeSelection={() =>{}} status="" isClosed={true} isWinner={false} selectedAttribute="" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(component.root.findAllByType(Card)).not.toBeNull();
    expect(component.root.findAllByType(SelectedAttribute)).not.toBeNull();
  });
});
