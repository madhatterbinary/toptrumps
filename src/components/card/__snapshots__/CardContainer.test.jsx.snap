// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer renders 1`] = `
<div
  className="card-container false"
>
  <div
    className="card closed"
  />
  <div
    className="card-result"
  >
    You won!
  </div>
</div>
`;
